index.js

```diff
@@ -1,125 +1,211 @@
+      [t.identifier(localName)]
+    )
+  )
+}
+
+function makeNamespaceObject (t, localName, members) {
+  const properties = members.map(member => t.objectProperty(
+    t.identifier(member.name), t.identifier(`_${localName}_${member.name}`)
+  ))
+  return t.variableDeclaration(
+    'const', [
+      t.variableDeclarator(
+        t.identifier(localName),
+        t.objectExpression(properties)
+      )
+    ]
+  )
 }
 
 module.exports = babelCore => {
   const t = babelCore.types
-
   return {
     visitor: {
-      ImportDeclaration (path, file) {
-        const node = path.node
-        if (!node.source.value.startsWith('glob:') && !glob.hasMagic(node.source.value)) {
-          return
-        }
+      ImportDeclaration (ast, state) {
+        const specifiers = ast.node.specifiers
+        const source = ast.node.source
+        const error = message => ast.buildCodeFrameError(message)
 
-        rejectImportDefaultSpecifier(path, node.specifiers)
-
-        const fromDir = nodePath.dirname(file.file.opts.filename)
-        const matches = glob.sync(getPattern(path, node.source), {
-          // Search relative to the source file, assuming that location is
-          // derived correctly.
-          cwd: fromDir,
-          strict: true
-        })
-
-        const prefix = commonPathPrefix(matches)
-        const suffix = commonExtname(matches)
-
-        const lookup = Object.create(null)
-        const members = []
-        for (const filepath of matches) {
-          const src = './' + nodePath.relative(fromDir, nodePath.resolve(fromDir, filepath))
-          const subpath = filepath.slice(prefix.length, filepath.length - suffix.length)
-          const pieces = subpath.split(nodePath.sep)
-          const member = pieces.map((name, index) => {
-            const id = identifierfy(name, {
-              prefixReservedWords: pieces.length === 1,
-              prefixInvalidIdentifiers: index === 0
-            })
-            if (id === null) {
-              throw path.buildCodeFrameError(
-                `Could not generate a valid identifier for '${src}'. The '${name}' component could not be converted.`)
-            }
-            return id
-          }).join('$')
+        let pattern = source.value
 
-          if (lookup[member]) {
-            // hyphen conversion means foo-bar and fooBar will collide.
-            throw path.buildCodeFrameError(`Found colliding members '${member}'`)
+        if (!glob.hasMagic(pattern)) {
+          if (pattern.startsWith('glob:')) {
+            throw error(`Missing glob pattern '${pattern}'`)
           }
-
-          lookup[member] = src
-          members.push(member)
+          return
         }
 
-        if (node.specifiers.length === 0) {
-          path.replaceWithMultiple(members.map(member => t.importDeclaration([], t.stringLiteral(lookup[member]))))
-          return
+        if (pattern.startsWith('glob:')) {
+          pattern = pattern.replace(/^glob:/, '')
```

Comment:
This could possibly be:
```js
pattern = pattern.substr('glob:'.length)
```
`glob:` could be changed to a constant. I think the regexp might be a tad overkill here.